### ISSUE TITLE
Style/SymbolArray autocorrect breaks code in partial-newline case

### ISSUE THREAD

**Issue by SaladFork** [bbatsov/rubocop#5686]
The `Style/SymbolArray` cop has autocorrect functionality to change, e.g., `[:one, :two]` to `%i(one two)`. This mostly works correctly, though there is one case where the autocorrect would change the functionality of the code. There is [already a test case](https://github.com/bbatsov/rubocop/blob/84156/spec/rubocop/cop/style/symbol_array_spec.rb#L39-L42) that tests for a similar case, though unfortunately does not catch this.

The existing test case (passes):
```rb
new_source = autocorrect_source("[:one,\n:two, :three,\n:four]")
expect(new_source).to eq("%i(one\ntwo three\nfour)")
```

A slightly modified version of the test case (fails on `master` at time of writing):
```rb
new_source = autocorrect_source("[:one, :two,\n:three,\n:four]")
expect(new_source).to eq("%i(one two\nthree\nfour)")
```

Test output:
```diff
expected: "%i(one two\nthree\nfour)"
     got: "%i(onetwo\nthree\nfour)"

(compared using ==)

Diff:
@@ -1,4 +1,4 @@
-%i(one two
+%i(onetwo
three
four)
```

I caught this because we had code similar to the following in our codebase:
```rb
SOME_CONSTANT = [:one, :two,
  :three,
  :four]
```

Which was incorrectly autocorrected (see expected/actual below).

--------

## Expected behavior

Rubocop should autocorrect into an array of the same symbols (preferably with the same or further autocorrected layout/indentation):
```rb
SOME_CONSTANT = %i(one two
  three
  four)
```

## Actual behavior

Rubocop merged the first two symbols:
```rb
SOME_CONSTANT = %i(onetwo
  three
  four)
```

## Steps to reproduce the problem

Add the above test to `spec/rubocop/cop/style/symbol_array_spec.rb` and observe it fail, or otherwise try to autocorrect the case provided above.

## RuboCop version

```
$ rubocop -V
0.53.0 (using Parser 2.5.0.0, running on ruby 2.4.1 x86_64-darwin17)
```

Tested via spec against `master` at commit `841569b30` (latest at time of writing).

**Comment by benmanns** [comment]
I ran into this as well. It seems like 0.49.1 does not have the issue. I'll take a look at bisecting where this was introduced.

**Comment by benmanns** [comment]
I narrowed it down to 2ee27d2a55d44c973dc8c6eb57a8bec76b7b769e.

**Comment by benmanns** [comment]
This is also present in Style/WordArray.

```sh
$ cat test.rb 
foo = ['test', '123',
       'foo', 'bar']
$ rubocop -a --only Style/WordArray test.rb 
Inspecting 1 file
C

Offenses:

test.rb:1:7: C: [Corrected] Style/WordArray: Use %w or %W for an array of words.
foo = ['test', '123', ...
      ^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
$ cat test.rb
foo = %w[test123
       foo bar]
```
